lib/checkunusedvar.cpp

```diff
@@ -1586,3 +1589,61 @@ bool CheckUnusedVar::isEmptyType(const Type* type)
     emptyType=false;   // unknown types are assumed to be nonempty
     return emptyType;
 }
+
+bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func)
+{
+    // no body to analyze
+    if (!func.hasBody() || !func.functionScope || !func.functionScope->bodyStart) {
```

Comment:
I believe that `hasBody` is only true if there is a `bodyStart`

Reply:
I added this check because I'm dereferencing `func.functionScope->bodyStart` few lines later - just to be sure that no segfault occurs. Do you mean that
```
if (!func.hasBody()) {
    return false;
}
```
is enough?

Reply:
yes I believe that is enough